atomix/cluster/src/test/java/io/atomix/raft/RaftRule.java

```diff
@@ -536,21 +530,27 @@ public void copySnapshotOffline(final String sourceNode, final String targetNode
     receivedSnapshot.persist();
   }
 
-  private RaftStorage createStorage(
-      final MemberId memberId,
-      final Function<RaftStorage.Builder, RaftStorage.Builder> configurator) {
-
+  private RaftStorage createStorage(final MemberId memberId, final Configurator configurator) {
     final var memberDirectory = getMemberDirectory(directory, memberId.toString());
-    final RaftStorage.Builder defaults =
+    final var snapshotStore = new TestSnapshotStore(getOrCreatePersistedSnapshot(memberId.id()));
+    snapshotStores.put(memberId.id(), snapshotStore);
+
+    if (configurator instanceof final SnapshotStoreConfigurator s) {
```

Comment:
🔧 It feels somehow wrong to use these instanceof 😅 couldn't we have different objects with the specific interfaces or all implementing/overloading/extending the same configure method ? 🤔


Or does this not work because we have to call it in different places?

Reply:
Or how about one interface with three default methods? Which can be implemented by the given configurators?

Reply:
The main difference with this approach is you can pass a lambda which implements any of the interfaces, or a class which implements multiple. If you have different objects, then you end up with many params and/or overloaded methods. If you have a single interface with multiple overloaded configure methods, you can't use a lambda (or maybe you can, but only in one case).

At any rate, I don't care that much to be honest - I figured it was fine since this is just test code, and I would prioritize ergonomy over performance or the likes, but :shrug: Happy to change it :+1: